common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/PipelineStorageStreamTest.scala

```diff
@@ -260,6 +261,37 @@ class PipelineStorageStreamTest
     }
   }
 
+  it(
+    "does not delete the message if the processFunction throws an exception not wrapped in a future") {
+    val index = createIndex
+    val document = SampleDocument(1, createCanonicalId, randomAlphanumeric())
+
+    val sender = new MemoryMessageSender
+
+    withLocalSqsQueuePair(visibilityTimeout = 10) {
+      case QueuePair(queue, dlq) =>
+        withPipelineStream(
+          queue = queue,
+          indexer = indexer(index),
```

Comment:
Could you use a MemoryIndexer instead, save having to spin up and spin down an Elasticsearch index?